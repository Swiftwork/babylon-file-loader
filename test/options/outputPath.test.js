/* eslint-disable
  prefer-destructuring,
*/
import webpack from '../helpers/compiler';

describe('Options', () => {
  describe('outputPath', () => {
    test('{String}', async () => {
      const config = {
        loader: {
          test: /\.babylon$/,
          options: {
            outputPath: '/test/',
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[1];

      expect(source).toMatchSnapshot();
    });

    test('{Function}', async () => {
      const config = {
        loader: {
          test: /\.babylon$/,
          options: {
            outputPath(url) {
              return `test/${url}`;
            },
          },
        },
      };

      const stats = await webpack('fixture.js', config);
      const { source } = stats.toJson().modules[1];

      expect(source).toMatchSnapshot();
    });
  });
});
