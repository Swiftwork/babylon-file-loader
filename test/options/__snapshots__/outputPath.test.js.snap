// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options outputPath {Function} 1`] = `"module.exports = __webpack_public_path__ + \\"test/77f94cfeb835c099e9eeab739989952d.babylon\\";"`;

exports[`Options outputPath {String} 1`] = `"module.exports = __webpack_public_path__ + \\"/test/77f94cfeb835c099e9eeab739989952d.babylon\\";"`;
